src/schemaCompare/schemaCompareWebViewController.ts

```diff
@@ -0,0 +1,320 @@
+    SchemaCompareWebViewState,
+} from "../sharedInterfaces/schemaCompare";
+import { TreeNodeInfo } from "../objectExplorer/treeNodeInfo";
+import ConnectionManager from "../controllers/connectionManager";
+import { IConnectionProfile } from "../models/interfaces";
+import {
+    cancel,
+    compare,
+    generateScript,
+    includeExcludeNode,
+    openScmp,
+    publishDatabaseChanges,
+    publishProjectChanges,
+    saveScmp,
+} from "./schemaCompareUtils";
+// import { ConnectionProfile } from "../models/connectionProfile";
+
+export class SchemaCompareWebViewController extends ReactWebviewPanelController<
+    SchemaCompareWebViewState,
+    SchemaCompareReducers
+> {
+    constructor(
+        context: vscode.ExtensionContext,
+        node: any,
+        private readonly schemaCompareService: mssql.ISchemaCompareService,
+        private readonly connectionMgr: ConnectionManager,
+        defaultDeploymentOptions: mssql.DeploymentOptions,
+        title: string,
+    ) {
+        super(
+            context,
+            "schemaCompare",
+            {
+                defaultDeploymentOptions: defaultDeploymentOptions,
+                sourceEndpointInfo: undefined,
+                targetEndpointInfo: undefined,
+                schemaCompareResult: undefined,
+                generateScriptResultStatus: undefined,
+                publishDatabaseChangesResultStatus: undefined,
+                schemaComparePublishProjectResult: undefined,
+                schemaCompareIncludeExcludeResult: undefined,
+                schemaCompareOpenScmpResult: undefined,
+                saveScmpResultStatus: undefined,
+                cancelResultStatus: undefined,
+            },
+            {
+                title: title,
+                viewColumn: vscode.ViewColumn.Active,
+                iconPath: {
+                    dark: vscode.Uri.joinPath(
+                        context.extensionUri,
+                        "media",
+                        "schemaCompare_dark.svg", // lewissanchez TODO - update icon for edit data
+                    ),
+                    light: vscode.Uri.joinPath(
+                        context.extensionUri,
+                        "media",
+                        "schemaCompare_light.svg", // lewissanchez TODO - update icon for edit data
+                    ),
+                },
+            },
+        );
+
+        if (!this.isTreeNodeInfoType(node)) {
+            node = this.getFullSqlProjectsPathFromNode(node);
+        }
+
+        void this.start(node);
+        this.registerRpcHandlers();
+    }
+
+    // schema compare can get started with four contexts for the source:
+    // 1. undefined
+    // 2. connection profile
+    // 3. dacpac
+    // 4. project
+    public async start(
+        sourceContext: any,
+        targetContext: mssql.SchemaCompareEndpointInfo = undefined,
+        comparisonResult: mssql.SchemaCompareResult = undefined,
+    ): Promise<void> {
```

Comment:
I'd break this up into some helper methods for each sourceContext type check so it'll be easier to read and test